src/source.rs

```diff
@@ -191,28 +190,17 @@ pub struct Message {
     pub keys: Vec<String>,
 }
 
-/// start_uds_server starts a gRPC server over an UDS (unix-domain-socket) endpoint.
+/// Starts a gRPC server over an UDS (unix-domain-socket) endpoint.
 pub async fn start_uds_server<T>(m: T) -> Result<(), Box<dyn std::error::Error>>
 where
-    T: Sourcer + Send + Sync + 'static,
+    T: Sourcer + Clone + Send + Sync + 'static,
```

Comment:
Allows `Arc<Sourcer>` to be created at user's side and reusing the same. Reduces one possible heap allocation if user also has to create an `Arc<Sourcer>`.